From 6d12ab77fa4681514637dfee8c707f6e6ea600d7 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Tue, 10 Apr 2018 08:04:54 +0800
Subject: [PATCH 1/5] ethernet: xilinx: TSN: fix compiling error when uprev to
 4.12

It seems module and of related header dependencies have been re-constructed, so
we explicitly include these two header to avoid the following errors:

drivers/net/ethernet/xilinx/xilinx_tsn_switch.c:67:1: warning: data definition has no type or storage class
 MODULE_DEVICE_TABLE(of, tsnswitch_of_match);
 ^~~~~~~~~~~~~~~~~~~
drivers/net/ethernet/xilinx/xilinx_tsn_switch.c:67:1: error: type defaults to 'int' in declaration of 'MODULE_DEVICE_TABLE' [-Werror=implicit-int]
drivers/net/ethernet/xilinx/xilinx_tsn_switch.c:67:1: warning: parameter names (without types) in function declaration
In file included from include/linux/dmaengine.h:20:0,
                 from include/linux/netdevice.h:38,
                 from drivers/net/ethernet/xilinx/xilinx_axienet.h:11,
                 from drivers/net/ethernet/xilinx/xilinx_tsn_switch.h:21,
                 from drivers/net/ethernet/xilinx/xilinx_tsn_switch.c:18:
include/linux/device.h:1477:1: warning: data definition has no type or storage class
 module_init(__driver##_init); \
 ^
include/linux/platform_device.h:228:2: note: in expansion of macro 'module_driver'
  module_driver(__platform_driver, platform_driver_register, \
  ^~~~~~~~~~~~~
drivers/net/ethernet/xilinx/xilinx_tsn_switch.c:791:1: note: in expansion of macro 'module_platform_driver'
 module_platform_driver(tsnswitch_driver);
 ^~~~~~~~~~~~~~~~~~~~~~
include/linux/device.h:1477:1: error: type defaults to 'int' in declaration of 'module_init' [-Werror=implicit-int]
 module_init(__driver##_init); \
 ^
include/linux/platform_device.h:228:2: note: in expansion of macro 'module_driver'
  module_driver(__platform_driver, platform_driver_register, \
  ^~~~~~~~~~~~~
drivers/net/ethernet/xilinx/xilinx_tsn_switch.c:791:1: note: in expansion of macro 'module_platform_driver'
 module_platform_driver(tsnswitch_driver);
 ^~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_tsn_switch.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/xilinx/xilinx_tsn_switch.c b/drivers/net/ethernet/xilinx/xilinx_tsn_switch.c
index cc7c205..db506f2 100644
--- a/drivers/net/ethernet/xilinx/xilinx_tsn_switch.c
+++ b/drivers/net/ethernet/xilinx/xilinx_tsn_switch.c
@@ -16,6 +16,8 @@
  */
 
 #include "xilinx_tsn_switch.h"
+#include <linux/of.h>
+#include <linux/module.h>
 #include <linux/of_platform.h>
 #include <linux/miscdevice.h>
 
-- 
1.8.5.2.233.g932f7e4

